rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -88,6 +88,19 @@ class Publisher : public PublisherBase
       this->add_event_handler(
         options_.event_callbacks.incompatible_qos_callback,
         RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+    } else {
+      // Register default callback when not specified
+      try {
+        this->add_event_handler(
+          QOSOfferedIncompatibleQoSCallbackType(default_incompatible_qos_callback),
+          RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+      } catch (UnsupportedEventTypeException & /*exc*/) {
+        RCLCPP_WARN_ONCE(
+          rclcpp::get_logger("rclcpp::Publisher"),
+          "This rmw implementation does not support ON_OFFERED_INCOMPATIBLE_QOS "
+          "events, you will not be notified when Publishers offer an incompatible "
+          "QoS profile to Subscriptions on the same topic.");
```

Comment:
I don't like to log a warning knowing that this will always be shown with `fastrtps` (until they support this events).

Reply:
Why?

We can add a way to suppress this warning, either programmatically or via an environment variable.

@mm318 Does fastrtps not support this already?

Reply:
Correct, @wjwwood, Fast-RTPS currently still does not support this type of callback (tracked at https://github.com/eProsima/Fast-RTPS/issues/852).

Reply:
> We can add a way to suppress this warning, either programmatically or via an environment variable.

Is there a standard way in ROS2 to suppress warnings?

Reply:
Not really no.

Reply:
This will essentially be controlled by `options_.use_default_callbacks`.